elasticsearch/templates/_helpers.tpl

```diff
@@ -21,7 +21,7 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 
 {{- define "masterService" -}}
 {{- if empty .Values.masterService -}}
-{{ .Values.clusterName }}-master
+{{ .Values.clusterName }}-{{ .Values.nodeGroup }}
```

Comment:
This would be a breaking change for anyone running clusters with multiple nodes types and using the default settings. One example is our [multi example](https://github.com/elastic/helm-charts/tree/master/elasticsearch/examples/multi). So if you are having a problem trying to setup a multi node cluster I would recommend opening an issue with the problems you are running into. 


How and why this is works is explained in more detail in [clustering-and-node-discovery](https://github.com/elastic/helm-charts/tree/master/elasticsearch#clustering-and-node-discovery) in the readme.

Reply:
Hello @Crazybus, the example only does work due to `nodeGroup: "master"` inside the multi example.

If you change this to `nodeGroup: "foobar"`, it will stop working due to ElasticSearch will will search for a `discovery.zen.ping.unicast.hosts = multi-master` and not `discovery.zen.ping.unicast.hosts = multi-foobar`

Reply:
Correct. As documented in the readme you also need to set `masterService` if you are changing the default `nodeGroup` for your group of masters.

> masterService | Optional. The service name used to connect to the masters. You only need to set this if your master nodeGroup is set to something other than master. See Clustering and Node Discovery for more information.

And also in the clustering and node discovery section that I linked to you above:

> If your group of master nodes has the default nodeGroup: master then you can just add new groups of nodes with a different nodeGroup and they will automatically discover the correct master. If your master nodes have a different nodeGroup name then you will need to set masterService to $clusterName-$masterNodeGroup.

If the documentation is unclear then we want to improve the documention. You proposed change would be a breaking change for everyone running multi node clusters with the default settings.

Reply:
Damn, you are right. Ok. Thanks for the clarification!